src/utils/indexer.js

```diff
@@ -248,18 +273,25 @@ const getAllTokenHolders = async () => {
     const result = await response.json();
     const tokens = result[0].children.find(elm => elm.name === 'ledger');
 
-    // const ledgerMetadata = await fetch(
-    //   `https://api.better-call.dev/v1/bigmap/${INDEXER_NETWORK}/${tokens.value}`,
-    // );
-    // const ledgerMetadataJSON = await ledgerMetadata.json();
-
-    const tk = await fetch(
-      `https://api.better-call.dev/v1/bigmap/${INDEXER_NETWORK}/${tokens.value}/keys`,
+    const ledgerMetadata = await fetch(
+      `https://api.better-call.dev/v1/bigmap/${INDEXER_NETWORK}/${tokens.value}`,
     );
+    const ledgerMetadataJSON = await ledgerMetadata.json();
+    const num_keys = ledgerMetadataJSON.active_keys;
+    const all_holders = [];
+    let tk;
+    for (let i = 0; i < parseInt(num_keys / 10) + 1; i++) {
+      tk = await fetch(
+        `https://api.better-call.dev/v1/bigmap/${INDEXER_NETWORK}/${
+          tokens.value
+        }/keys?offset=${10 * i}`,
+      );
+      all_holders.push(...(await tk.json()));
```

Comment:
```
for (let i = 0; i < parseInt(num_keys / 10) + 1; i++) {
      tk = await fetch(
        `https://api.better-call.dev/v1/bigmap/${INDEXER_NETWORK}/${
          tokens.value
        }/keys?offset=${10 * i}`,
      );
      all_holders.push(...(await tk.json()));
```

Please elaborate on what's going here. I think all fetches are being stacked here, right? Which would load all the bots eventually.

Reply:
Yep, spot on. 

We find out how many are the total keys, and then we know that page size is 10. 
So, we loop until we get all out bots, on every loop we fetch 10 bots with the correct offset.